packages/generator-clay/app/index.js

```diff
@@ -0,0 +1,153 @@
+        this.capitalizeName = _.startCase(componentName);
+        this.kebabCaseName = _.kebabCase(componentName);
+
+        this.repoName = this.kebabCaseName;
+        this.repoOwner = props.repoOwner;
+        this.repoDescription = props.repoDescription;
+        done();
+      }.bind(this)
+    );
+  },
+
+  writing: function() {
+    this.destinationRoot(this.repoName);
+    this.fs.copyTpl(
+      this.templatePath('demos/_index.html'),
+      this.destinationPath('demos/index.html'),
+      {
+        camelCaseName: this.camelCaseName,
+        componentName: this.componentName,
+        capitalizeName: this.capitalizeName,
+        kebabCaseName: this.kebabCaseName,
+        repoName: this.repoName,
+      }
+    );
+    this.fs.copyTpl(
+      this.templatePath('src/_Boilerplate.soy'),
+      this.destinationPath('src/' + this.componentName + '.soy'),
+      {
+        componentName: this.componentName,
+        kebabCaseName: this.kebabCaseName,
+      }
+    );
+    this.fs.copyTpl(
+      this.templatePath('src/_Boilerplate.js'),
+      this.destinationPath('src/' + this.componentName + '.js'),
+      {
+        buildFormat: this.buildFormat,
+        componentName: this.componentName,
+        templateLanguage: this.templateLanguage,
+        kebabCaseName: this.kebabCaseName,
+      }
+    );
+
+    this.fs.copyTpl(
+      this.templatePath('src/__tests__/_Boilerplate.js'),
+      this.destinationPath('src/__tests__/' + this.componentName + '.js'),
+      {
+        componentName: this.componentName,
+        kebabCaseName: this.kebabCaseName,
+      }
+    );
+    this.fs.copyTpl(
+      this.templatePath('src/__tests__/__snapshots__/_Boilerplate.js.snap'),
+      this.destinationPath(
+        'src/__tests__/__snapshots__/' + this.componentName + '.js.snap'
+      ),
+      {
+        componentName: this.componentName,
+        kebabCaseName: this.kebabCaseName,
+      }
+    );
+    this.fs.copyTpl(
+      this.templatePath('_package.json'),
+      this.destinationPath('package.json'),
+      {
+        componentName: this.componentName,
+        repoName: this.repoName,
+        repoOwner: this.repoOwner,
+        repoDescription: this.repoDescription,
+      }
+    );
+    this.fs.copyTpl(
+      this.templatePath('_README.md'),
+      this.destinationPath('README.md'),
+      {
+        repoName: this.repoName,
+        repoDescription: this.repoDescription,
+        componentName: this.componentName,
+        kebabCaseName: this.kebabCaseName,
+      }
+    );
+    this.fs.copy(
+      this.templatePath('_LICENSE.md'),
+      this.destinationPath('LICENSE.md')
+    );
+    this.fs.copyTpl(
+      this.templatePath('_webpack.config.js'),
+      this.destinationPath('webpack.config.js'),
+      {
+        componentName: this.componentName,
+        kebabCaseName: this.kebabCaseName,
+        repoName: this.repoName,
+      }
+    );
+  },
+
+  install: function() {
+    this.npmInstall();
```

Comment:
We will have to add `this.spawnCommand('yarn', ['install'])` to work properly with the working puppy.